runtime/dart_isolate.h

```diff
@@ -416,8 +439,11 @@ class DartIsolate : public UIDartState {
   std::unordered_set<fml::RefPtr<DartSnapshot>> loading_unit_snapshots_;
   fml::RefPtr<fml::TaskRunner> message_handling_task_runner_;
   const bool may_insecurely_connect_to_all_domains_;
-  std::string domain_network_policy_;
+  const bool is_platform_isolate_;
   const bool is_spawning_in_group_;
+  std::string domain_network_policy_;
+  uint32_t platform_isolate_pending_messages_ = 0;
+  PlatformIsolateManager* platform_isolate_manager_ = nullptr;
```

Comment:
As multiple (UI) isolates may hold on to this pointer, it's somewhat unclear who maintains ownership and what prevents this from being accessed in use-after-free situation. Please add a comment.

Reply:
`PlatformIsolateManager` should be held in a `std::shared_ptr` within `RuntimeController`/`DartIsolate`.  It should be captured as a `std::weak_ptr` in the tasks queued to the platform thread.

If engine shutdown happens before a platform thread task executes, then the `RuntimeController` will be destructed before the task runs.  The tasks queued by `DartIsolate::CreatePlatformIsolate` can detect this safely with a `weak_ptr`.

Reply:
Done.